File: GreenPotFront/src/app/dto/plantAllDataDto.ts
import {Plant} from "../models/plant";
import {SubCategory} from "../models/subCategory";
import {Image} from "../models/image";
import {Calendar} from "../models/calendar";

export interface PlantAllDataDto{
  plant: Plant;
  subCategoriesList: SubCategory[];
  imagesList: Image[];
  calendarList: Calendar[];
}

File: GreenPotBack/src/main/java/com/example/greenpotback/Dto/PlantAllDataDto.java
package com.example.greenpotback.Dto;

import com.example.greenpotback.Plant.Calendar.Calendar;
import com.example.greenpotback.Plant.Image.Image;
import com.example.greenpotback.Plant.Plant;
import com.example.greenpotback.Plant.SubCategory.SubCategory;
import lombok.Data;

import java.util.ArrayList;
import java.util.List;

@Data
public class PlantAllDataDto {
    Plant plant;
    List<SubCategory> subCategoriesList;
    List<Image> imagesList;
    List<Calendar> calendarList;
}
